tests/test_data/test_scrape.py

```diff
@@ -0,0 +1,62 @@
+import os
+
+from typing import List
+
+import pytest
+
+from data.src.scrape import _build_scraping_config
+
+from . import handle_dummy_files
+
+
+# "mul" should be a future-proof iso639 code to test with.
+# "mul" is resolved to "Multiple Languages" by iso639 package,
+# which is a non-existent category on Wikitionary.
+# An alternative solution to using "mul" would be to add
+# a code to languagecodes.py explicitly for the purposes of testing.
+@pytest.mark.parametrize(
+    "config_settings, dialect_suffix, phones, expected_file_name",
+    [
+        # Dialect and phones
+        (
+            {"key": "mul"},
+            "test_",
+            True,
+            [
+                "mul_test_phonetic.tsv",
+                "mul_test_phonemic.tsv",
+                "mul_test_phonetic_filtered.tsv",
+                "mul_test_phonemic_filtered.tsv",
+            ],
+        ),
+        # Dialect
+        (
+            {"key": "mul"},
+            "test_",
+            False,
+            ["mul_test_phonetic.tsv", "mul_test_phonemic.tsv"],
+        ),
+        # Standard
+        ({"key": "mul"}, "", False, ["mul_phonetic.tsv", "mul_phonemic.tsv"],),
+    ],
+)
+def test_file_creation(
+    config_settings: object,
+    dialect_suffix: str,
+    phones: bool,
+    expected_file_name: List[str],
+):
+    """Check whether _build_scraping_config() outputs TSVs with expected
+    file names based on presence or absence of dialect specification
+    or .phones files for a given language.
+    """
+    with handle_dummy_files(
+        phones, config_settings["key"], dialect_suffix
+    ) as dummy_tsv_path:
+        _build_scraping_config(
+            config_settings=config_settings, dialect_suffix=dialect_suffix
+        )
+        tsv_contents = os.listdir(dummy_tsv_path)
```

Comment:
I looked at using `tempfile`, but the problem I ran into was that `_build_scraping_config()` in src/scrape.py is using the following to define its paths: 
```
    path_affix = f”../tsv/{config_settings[‘key’]}_{dialect_suffix}”
    phones_path_affix = f"../phones/{config_settings['key']}_{dialect_suffix}"
```
Meaning running `build_scraping_config()` from `test_data` results in it looking for `tsv` and `phones` directories in `tests`. I can’t see a way of using `tempfile` that doesn’t also involve changing the current working directory, as in the following crude solution:
```
    with tempfile.TemporaryDirectory() as temp_dir:
        os.mkdir(f"{temp_dir}/tsv")
        os.mkdir(f"{temp_dir}/phones")
        if phones:
            write_dummy_phones_files(
                f"{temp_dir}/phones",
                config_settings["key"],
                dialect_suffix
            )
        os.chdir(f"{temp_dir}/tsv") # <----- Awkwardly change cwd
        _build_scraping_config(
            config_settings=config_settings, dialect_suffix=dialect_suffix
        )
        os.chdir(_CURRENT_DIR)
```
Hopefully these comments shed some light on the pickle I’ve found myself in.

Reply:
FWIW I think the `chdir` here isn't *that bad*, but I'm not sure that using `tempfile` is good enough to make it worth it. @jacksonllee what do you think?

Reply:
Ah, now I see what's happening -- I haven't been as familiar with the big scrape code as you both and I was just looking at the `_build_scraping_config` source more closely now. If the issue is about the hard-coded `path_affix` and `phones_path_affix` defined inside `_build_scraping_config`, then it might suggest this sort of affixes/paths could be passed into `_build_scraping_config` somehow instead. This being said, refactoring is probably out of scope for this PR.

I'd imagine `tempfile.TemporaryDirectory` would be preferred over a homegrown context manager for the same purpose, but using `os.chdir` like Lucas sketched here would seem odder than what has already been committed in code in this PR. I'm LGTM-ing now. :-) As more tests are developed, they will bring us more thoughts about how we might restructure things a little bit to avoid less common patterns.